MdeModulePkg/Core/Dxe/DxeMain/DxeMain.c

```diff
@@ -260,10 +261,17 @@ DxeMain (
   //
   // Setup Stack Guard
   //
-  // MU_CHANGE START
+  // MU_CHANGE START: Check Memory Protection HOB
   // if (PcdGetBool (PcdCpuStackGuard)) {
-  //  Status = InitializeSeparateExceptionStacks (NULL, NULL);
-  //  ASSERT_EFI_ERROR (Status);
+  Ptr = GetFirstGuidHob (&gDxeMemoryProtectionSettingsGuid);
+  if ((Ptr != NULL) &&
+      (*((UINT8 *)GET_GUID_HOB_DATA (Ptr)) == (UINT8)DXE_MEMORY_PROTECTION_SETTINGS_CURRENT_VERSION) &&

```

Comment:
And this would be more readable if you use the field names and structure types.

Reply:
I agree. It would also be a bit less error prone in case of a non-backward compatible structure change in the future (which shouldn't happen anyway).

Reply:
Updated